### PR TITLE
[Documentation-Plugin]: rework of the plugin and bugfix

### LINKED ISSUE #186: Documentation Plugin Download White Screen

### PR DESCRIPTION
**Pull request by michelguerin**
# 🗒 Description

The documentation plugin was not working : blank pages opened on button click

## 📷 Demo screen recording or Screenshots

With the biggest sample file that I have :

[ce-export.pdf](https://github.com/user-attachments/files/17848211/ce-export.pdf)

[list-elements.pdf](https://github.com/user-attachments/files/17848212/list-elements.pdf)

Screenshot 2024-11-21 at 16 22 14

Screenshot 2024-11-21 at 16 22 29

## 📋 Checklist

You may remove tasks that do not make sense in this PR.

- [x] Ticket-ACs are checked and met
- [x] GitHub **labels** are assigned
- [x] Git Ticket / issue  is linked with PR
- [ ] Designated PR Reviewers are set
- [ ] Tested by another developer
- [x] Changelog updated
- [ ] All comments in the PR are resolved / answered

## Linked to

Closes #186, closes #185

## Usefull commits 

* Migration from Use case to domain driven: 050a52652d9712d879054f427006914da585a082
* Adding editCount to the plugin instance for reactivity: f214d69498355d83e88442cd123949b75cbc14a5
* Communication Explorer refactor after migration : 2dd9323d3280402dd1fa24c4957b9391e8125178
* Rework and simplification of documentation plugin : e3ea4a7a6f762dfaa7a4651a8d812934168e2553

### CODE REVIEW

**Comment by michelguerin** [comment]
@LissiEitel Could you review the UI part (I'll attach the pdf samples) ?

**Comment by LissiEitel** [comment]
> @LissiEitel Could you review the UI part (I'll attach the pdf samples) ?

Hi Michel i looked at it. I am not sure how it was in the past, but i think it would be better if all lists would be arranged one below the other. @PhilipLeddin what do you think? How was it in the past?

**Comment by PhilipLeddin** [comment]
@michelguerin @LissiEitel , it was one below the other in the past. I think that would be better when you think of longer names and so on.

File: packages/core/src/scd-queries/ied/queries.ied.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		)
+	}
+
+	public static SelectorExtRef = SCD_ELEMENTS.extRef.element.tag
+	public static AttributeListExtRef: Array<keyof ExtRefStandardAttributes> = [
+		'iedName',
+		'serviceType',
+		// "ldInst",
+		// "lnClass",
+		// "lnInst",
```
Comment (salvar3nga) [inline packages/core/src/scd-queries/ied/queries.ied.ts:66]: Is there a reason for leaving them commented out?

File: packages/core/src/scd-queries/ied/service.ied.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+	//====== PUBLIC METHODS
+
+	public IEDCommunicationInfos(): IED.CommunicationInfo[] {
+		const ieds = this.iedQueries.searchIEDs()
+		const communicationInfos: IED.CommunicationInfo[] = ieds.map((ied) => {
+			return {
+				iedName: ied.name,
+				iedDetails: this.parseDetails(ied.element),
+				// published: this.findPublishedMessages(ied),
```
Comment (salvar3nga) [inline packages/core/src/scd-queries/ied/service.ied.ts:28]: also here?

File: packages/core/src/scd-queries/ied/types.ied.d.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		id: string
+		name: string
+		targetIEDName: string
+		serviceType: string
+		serviceDatSet: string
+		serviceCbName: string
+	}
+
+	export type ReceivedMessage = {
+		iedName: string // to show
```
Comment (salvar3nga) [inline packages/core/src/scd-queries/ied/types.ied.d.ts:40]: to show = is displayed on the page?
to filter = you use it to search?

File: packages/uilib/src/lib/plugins/communication-explorer/telemetry-view/telemetry-view.svelte [hunk 4]
```diff
@@ -1,73 +1,85 @@
...
+		selectedFilter,
+		preferences
+	)
+}
+
+const config: Config = {
+	width: 150,
+	height: 40
+	// spacingBetweenNodes: 100,
+	// spacingBase: 40,
```
Comment (salvar3nga) [inline packages/uilib/src/lib/plugins/communication-explorer/telemetry-view/telemetry-view.svelte:65]: can be removed if not needed oder?

**Review by salvar3nga — COMMENTED** [review]
Looks good to me. Just left a few questions here and there

File: packages/core/src/scd-queries/ied/queries.ied.ts (hunk 1 above)
Comment (michelguerin) [reply packages/core/src/scd-queries/ied/queries.ied.ts:66]: This comes from the legacy code, I forgot to clean them ;)

File: packages/core/src/scd-queries/ied/types.ied.d.ts (hunk 3 above)
Comment (michelguerin) [reply packages/core/src/scd-queries/ied/types.ied.d.ts:40]: Like the other comments, these comes from the legacy codebase. I'll remove them as they are confusing.

File: packages/uilib/src/lib/plugins/communication-explorer/telemetry-view/telemetry-view.svelte (hunk 4 above)
Comment (michelguerin) [reply packages/uilib/src/lib/plugins/communication-explorer/telemetry-view/telemetry-view.svelte:65]: I think you are right, but this plugin is out of my scope: I have updated the plugin with the new core features, and I would rather avoid making any other changes for now.
So I will not update it and I will leave it to @interBN or @pwinkelholzS1 if they are working on this side of the plugin.

**Comment by michelguerin** [comment]
> > @LissiEitel Could you review the UI part (I'll attach the pdf samples) ?
> 
> Hi Michel i looked at it. I am not sure how it was in the past, but i think it would be better if all lists would be arranged one below the other. @PhilipLeddin what do you think? How was it in the past?

Yep thanks for the catch : that way a layout mistake. I made the changes.

**Pull request merged**